parser/src/lexer.rs

```diff
@@ -456,6 +458,27 @@ where
         }
     }
 
+    fn uncicode_literal(&mut self, literal_number: usize) -> Result<char, LexicalError> {
+        let location = self.get_pos();
+        let mut it: Vec<u32> = Vec::new();
+        for _ in 1..=literal_number {
+            match self.next_char() {
+                Some(d) => it.push(d.to_digit(16).unwrap()),
+                None => {
+                    return Err(LexicalError {
+                        error: LexicalErrorType::UnicodeError,
+                        location: location,
+                    })
+                }
+            }
+        }
+        let mut p: u32 = 0u32;
+        for i in 1..=literal_number {
```

Comment:
In this case, I suggest to use `enumerate` to iterate over the parsed digits and construct the numerical value.
```rust
for (i, d) in enumerate it {
   p += d << i*4;
}
```

Reply:
Will the compiler optimize it out though?

Reply:
Anyway, we can even do something like this

```rust
let p = it.iter().enumerate().fold(|acc, (i, d) acc + d << i *4)
```
But only if this all line ends up being optimized to a loop

Reply:
The `it` is removed, but I learn the powerful `fold` method, I will try to use it next time.    Thank you. @michelhe